File: app/src/main/java/ru/org/adons/cblock/app/ApplicationComponent.java
package ru.org.adons.cblock.app;

import android.content.Context;

import com.squareup.sqlbrite.BriteContentResolver;

import javax.inject.Singleton;

import dagger.Component;
import ru.org.adons.cblock.data.DaoSession;

/**
 * Provide application scope dependencies (Context, Preferences, DaoSession)
 */
@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    Context context();

    Preferences preferences();

    BriteContentResolver resolver();

    DaoSession daoSession();

    BlockManager blockManager();

}

File: app/src/main/java/mobile/addons/cblock/app/ApplicationComponent.kt
package mobile.addons.cblock.app

import android.content.Context
import com.squareup.sqlbrite.BriteContentResolver
import dagger.Component
import mobile.addons.cblock.data.DaoSession
import javax.inject.Singleton

/**
 * Provide application scope dependencies (Context, Preferences, DaoSession)
 */
@Singleton
@Component(modules = [(ApplicationModule::class)])
interface ApplicationComponent {

    fun context(): Context

    fun preferences(): Preferences

    fun resolver(): BriteContentResolver

    fun daoSession(): DaoSession

    fun blockManager(): BlockManager

}
